File: src/main/java/com/vaadin/demo/component/breadcrumbs/BreadcrumbsIcons.java
package com.vaadin.demo.component.breadcrumbs;

import com.vaadin.flow.component.breadcrumbs.Breadcrumbs;
import com.vaadin.flow.component.breadcrumbs.BreadcrumbsItem;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.icon.VaadinIcon;
import com.vaadin.flow.router.Route;

import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("breadcrumbs-icons")
public class BreadcrumbsIcons extends Div {

    public BreadcrumbsIcons() {

        // tag::snippet[]
        Breadcrumbs breadcrumbs = new Breadcrumbs(Breadcrumbs.Mode.MANUAL);

        BreadcrumbsItem home = new BreadcrumbsItem("Home", "/",
                VaadinIcon.HOME.create());
        BreadcrumbsItem orders = new BreadcrumbsItem("Orders", "/orders",
                VaadinIcon.PACKAGE.create());
        BreadcrumbsItem current = new BreadcrumbsItem("Order Details");

        breadcrumbs.add(home, orders, current);
        // end::snippet[]

        breadcrumbs.getElement() // hidden-source-line
                .executeJs("window.patchBreadcrumbsNavigation(this);"); // hidden-source-line
        add(breadcrumbs);
    }

    public static class Exporter // hidden-source-line
            extends DemoExporter<BreadcrumbsIcons> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/breadcrumbs/breadcrumbs-icons.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/breadcrumbs';
import '@vaadin/icon';
import '@vaadin/icons';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { patchBreadcrumbsNavigation } from 'Frontend/demo/component/breadcrumbs/breadcrumbs-helper'; // hidden-source-line
import { applyTheme } from 'Frontend/demo/theme';

@customElement('breadcrumbs-icons')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  /* prettier-ignore */ protected firstUpdated() { // hidden-source-line
    patchBreadcrumbsNavigation(this.shadowRoot!.querySelector('vaadin-breadcrumbs')!); // hidden-source-line
  } // hidden-source-line

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-breadcrumbs>
        <vaadin-breadcrumbs-item path="/">
          <vaadin-icon icon="vaadin:home" slot="prefix"></vaadin-icon>
          Home
        </vaadin-breadcrumbs-item>
        <vaadin-breadcrumbs-item path="/orders">
          <vaadin-icon icon="vaadin:package" slot="prefix"></vaadin-icon>
          Orders
        </vaadin-breadcrumbs-item>
        <vaadin-breadcrumbs-item>Order Details</vaadin-breadcrumbs-item>
      </vaadin-breadcrumbs>
      <!-- end::snippet[] -->
    `;
  }
}
